homeassistant/components/climate/radiotherm.py

```diff
@@ -77,19 +111,25 @@ class RadioThermostat(ClimateDevice):
     def __init__(self, device, hold_temp, away_temps):
         """Initialize the thermostat."""
         self.device = device
+        # It would be better if this was in update() since it triggers
+        # a network call but the thermostat will clear any temporary
+        # mode or temperature if this is called so we have to leave it here.
         self.set_time()
```

Comment:
You could move this to the coroutine entity method `async_added_to_hass`, which is called once, when the entity is added to home assistant, and run `set_time` in the executor via `hass.async_add_job`. See eg:
https://github.com/home-assistant/home-assistant/blob/dev/homeassistant/components/lutron.py#L70-L72

Reply:
Thanks.